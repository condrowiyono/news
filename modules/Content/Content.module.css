.container {
  width: 100%;
}

.pillContainer {
  display: flex;
  justify-content: space-evenly;
  gap: 8px;
  padding-top: 16px;
  padding-bottom: 16px;
}

.listContainer {
  grid-template-columns: repeat(1, minmax(0, 1fr));
  display: grid;
  gap: 14px;
  justify-content: space-between;
}

.livestreamContainer, .quizContainer {
  display: grid;
  gap: 14px;
  justify-content: space-between;
  grid-template-columns: repeat( auto-fit, minmax(120px, 1fr) );
}