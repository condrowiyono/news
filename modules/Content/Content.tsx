import List, { SkeletonList } from "@components/List";
import Livestream, { SkeletonLivestream } from "@components/Livestream";
import Pill from "@components/Pill";
import Thumbnail, { SkeletonThumbnail } from "@components/Thumbnail";
import { useEffect, useState } from "react";
import style from "./Content.module.css";

const Tabs = [
  { key: "news", label: "Berita" },
  { key: "livestream", label: "Livestream" },
  { key: "quiz", label: "Quiz" },
];

interface IContent {
  data: INews[];
  isLoading?: boolean;
  onTabChange?: (category: string) => void;
}

const Content = ({ data, onTabChange, isLoading }: IContent) => {
  const [active, setActive] = useState("news");

  useEffect(() => {
    onTabChange?.(active);
  }, [active]);

  return (
    <div className={style.container}>
      <div className={style.pillContainer}>
        {Tabs.map((item) => (
          <Pill
            key={item.key}
            active={active === item.key}
            onClick={() => setActive(item.key)}
          >
            {item.label}
          </Pill>
        ))}
      </div>
      {active === "news" && (
        <div className={style.listContainer}>
          {isLoading
            ? [1, 2, 3, 4].map((el) => <SkeletonList />)
            : data.map((news) => <List key={news.id} {...news} />)}
        </div>
      )}
      {active === "livestream" && (
        <div className={style.livestreamContainer}>
          {isLoading
            ? [1, 2, 3, 4].map((el) => <SkeletonLivestream />)
            : data.map((news) => <Livestream key={news.id} {...news} />)}
        </div>
      )}
      {active === "quiz" && (
        <div className={style.quizContainer}>
          {isLoading
            ? [1, 2, 3, 4].map((el) => <SkeletonThumbnail />)
            : data.map((news) => <Thumbnail key={news.id} {...news} />)}
        </div>
      )}
    </div>
  );
};

export default Content;
