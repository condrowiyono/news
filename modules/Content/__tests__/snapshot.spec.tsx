import { render } from "@testing-library/react";
import Content from "@modules/Content";
import { news } from "./Content.spec";

it("renders Content unchanged", () => {
  const { container } = render(<Content data={news} />);
  expect(container).toMatchSnapshot();
});
