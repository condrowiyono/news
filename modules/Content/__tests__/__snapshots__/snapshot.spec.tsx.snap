// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Content unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="pillContainer"
    >
      <button
        class="pill active"
        data-testid="pill"
      >
        Berita
      </button>
      <button
        class="pill "
        data-testid="pill"
      >
        Livestream
      </button>
      <button
        class="pill "
        data-testid="pill"
      >
        Quiz
      </button>
    </div>
    <div
      class="listContainer"
    >
      <a
        class="container"
        href="/#"
      >
        <div
          class="header"
        >
          <div>
            <div
              class="date"
            >
              2 Agustus 2022 08.12
            </div>
            <div
              class="title"
              data-testid="title"
            >
              Samantha William
            </div>
          </div>
          <div
            class="thumbnail"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27112%27%20height=%2784%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Samantha William"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="border-radius: 8px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/gif;base64,R0lGODlhAQABAPAAAL29vf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==);"
              />
              <noscript />
            </span>
            <div
              class="watermark"
            >
              <img
                alt="logo-idn"
                height="16"
                src="/logos/logo_idntimes.png"
                width="45"
              />
            </div>
          </div>
        </div>
        <div
          class="footer"
        >
          <div
            class="category"
          >
            Bisnis
          </div>
          <button
            class="share"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="share-button"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/gif;base64,R0lGODlhAQABAPAAAL29vf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==);"
              />
              <noscript />
            </span>
          </button>
        </div>
      </a>
    </div>
  </div>
</div>
`;
